### PR TITLE
refactor: expose built-in engines to `sqllogictest-engines` crate

### LINKED ISSUE #153: consider exposing pre-built `engines` public
Hi, thanks for your awesome work! I'm a developer from RisingWave Labs 😄, and we're recently developing our customized simulation tester with the `sqllogictest` library.

RisingWave is a database that is mostly compatible with the PostgreSQL wire protocol. To implement a runner for the simulated RisingWave, we have to copy the code from `bin/source/engines/postgres.rs` and slightly wrap it with some other logic. This can actually be achieved by forwarding the implementation to the `Postgres as AsyncDb`, so I'm considering whether it's possible to expose this pre-built engine so that we can have better synchronization with the upstream?

If you don't mind, I can submit a PR to work on it. 🥰

### PR DESCRIPTION
**Pull request by BugenZhao**
In this PR, we add a new workspace member of `sqllogictest-engine`, which contains pre-built engines like `PostgresSimple`, `PostgresExtended`, and `ExternalDriver`. Developers can directly depend on the `sqllogictest-engine` crate to utilize them.

- Close #153

### CODE REVIEW

**Comment by BugenZhao** [comment]
cc @xxchan @wangrunji0408 @skyzh PTAL. 🥰

BTW, should I do something else for adding a new crate?

**Comment by BugenZhao** [comment]
It seems the file `postgres_extended_test.slt` is unused... 🤔

File: sqllogictest-engine/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[package]
+name = "sqllogictest-engine"
+version = { workspace = true }
```
Comment (xxchan) [inline sqllogictest-engine/Cargo.toml:3]: We can begin to use separated versions? 🤪

File: sqllogictest-engine/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+futures = { version = "0.3", default-features = false }
+log = "0.4"
+pg_interval = "0.4"
+postgres-types = { version = "0.2.3", features = ["derive", "with-chrono-0_4"] }
+rust_decimal = { version = "1.7.0", features = ["tokio-pg"] }
+serde = { version = "1", features = ["derive"] }
+serde_json = "1"
+sqllogictest = { path = "../sqllogictest", version = "0.11" }
+thiserror = "1"
+tokio = { version = "1", features = [
```
Comment (xxchan) [inline sqllogictest-engine/Cargo.toml:24]: Do we have to depend on `tokio` 🤔

File: sqllogictest-engine/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+[package]
+name = "sqllogictest-engine"
```
Comment (xxchan) [inline sqllogictest-engine/Cargo.toml:2]:
```suggestion
name = "sqllogictest-engines"
```
? 😄

File: sqllogictest-engine/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+version = { workspace = true }
+edition = { workspace = true }
+homepage = { workspace = true }
+keywords = { workspace = true }
+license = { workspace = true }
+repository = { workspace = true }
+description = "Sqllogictest pre-built engines."
```
Comment (xxchan) [inline sqllogictest-engine/Cargo.toml:9]:
```suggestion
description = "Sqllogictest built-in engines."
```
? 🤔

**Review by xxchan — APPROVED** [review]

File: sqllogictest-engine/Cargo.toml (hunk 1 above)
Comment (BugenZhao) [reply sqllogictest-engine/Cargo.toml]: I'm not sure, but it seems not bad for using a unified version 🤣 which is less ambiguous for users.

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**